doc/guides/west/index.rst

```diff
@@ -11,11 +11,16 @@ sub-command::
 
   west [common-opts] <command> [opts] <args>
 
-West's built-in commands provide a multiple repository management system with
-features inspired by Google's Repo tool and Git submodules. West is also
-pluggable: you can write your own west "extension commands" which add
-additional features to west.  Zephyr provides extension commands for building
-applications, debugging them, and more.
+West's built-in commands provide a multiple repository management
+system with features inspired by Google's Repo tool and Git
+submodules. West simplifies configuration and is also pluggable: you
+can write your own west "extension commands" which add additional
+features to west.  Zephyr uses this feature to provide conveniences
+for building applications, flashing and debugging them, and more.
+
+It is possible not to use west for Zephyr development if you do not
+require these features or prefer to use your own tools; however, this
+implies extra effort and expert knowledge.
```

Comment:
Nit: I know you're trying to keep this short but I'd like to a very brief mention like this:

> if you do not require these features, prefer to use your own tools or need to work on the tools `west` relies on;

As this is an open-source project I think it's an important "use-case".

BTW I think you could save on the earlier paragraph like this:

> West's built-in commands provide a multiple repository management system with features inspired by Google's Repo tool and Git submodules. West simplifies configuration and is also pluggable: some of its current "extension commands" provide conveniences for building applications, flashing and debugging them, and more.

There's a longer version further down.

Reply:
> West simplifies configuration and is also pluggable: some of its current "extension commands" provide conveniences for building applications, flashing and debugging them, and more.

The reason I didn't do something like that is that extensions, by definition, are not part of west, which "its [...] extension commands" may imply. Extension commands are a west feature. The build etc. extensions are zephyr features.

> Nit: I know you're trying to keep this short but I'd like to a very brief mention like this:

I assume you mean in the getting started guide? I've made a change there now.

Reply:
> I assume you mean in the getting started guide? I've made a change there now.

No I meant right here. I merely added 9 words to your own sentence. I think you read too fast and missed them.

Reply:
> I think you read too fast and missed them.

I didn't read too fast. I was just confused about what you meant, which is why I used the word "assume" and a question mark :stuck_out_tongue:.

The main document I don't want to make longer is the getting started guide.

> or need to work on the tools west relies on;

I'm not totally sure what you mean by this, but I don't think it's accurate as written. For example, if you're working on pyOCD, you don't need to use Zephyr without west. You could still use `west flash` with the pyocd runner and your own pyOCD fork.

I've added "or want to eliminate the extra layer of indirection" instead. Is that what you meant?

Reply:
> You could still use west flash with the pyocd runner and your own pyOCD fork.

Sure, however if you want to interactively invoke pyOCD with a variety of unusual debug/tracing options/hacks because you're trying to find out what's wrong in pyOCD itself then there is very often a point at which it becomes more convenient to skip west. Same when you're trying to isolate subtle environment related issues.

I'm not familiar with pyOCD but I'm familiar with too many layers of indirections just like everyone else so it was really easy to make this one up :-)

> I've added "or want to eliminate the extra layer of indirection" instead. Is that what you meant?

Perfect.